### DISCUSSION TITLE
Access files

### DISCUSSION THREAD

**Discussion by wosigor** [vercel/community#4479] [Help] [answered]
### Summary

I can't seem to access files from the project directory. Here is the outline of the project directory.

public
- erc20.txt
- invoice-minter.txt

src
- server.ts

Here is my vercel.json
```javascript
{
  "outputDirectory": "build",
  "buildCommand": "next build && cp erc20.txt output/erc20.txt && cp invoice-minter.txt output/invoice-minter.txt",
  "builds": [
    {
      "src": "src/server.ts",
      "use": "@vercel/node",
      "config": {
        "includeFiles": ["erc20.txt", "invoice-minter.txt"]
      }
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "src/server.ts"
    }
  ]
}
``` 

In server.ts, I'm trying to access the contents of the file:

```javascript
const ERC20_ADDRESS: string = fs.readFileSync("/public/erc20.txt", "utf8");
```

After deployment, when I'm calling a function, I get the following error:
```javascript
Error: ENOENT: no such file or directory, open '/public/erc20.txt'
```

Please help with how to properly upload files and access them from the server.ts file.
Thanks in advance!

### Steps to Reproduce

Error: ENOENT: no such file or directory, open 'erc20.txt'

**Comment by amyegan** [comment, marked as answer]
Hi @wosigor! The file path should start with the current working directory of the Node.js process. So if the `/public` directory is located at the root of the project, you could use something like `const file = path.join(process.cwd(), 'public', 'erc20.txt');` and pass that file path to `readFileSync`.

https://vercel.com/guides/how-can-i-use-files-in-serverless-functions
